src/Bootstrap/Base.php

```diff
@@ -134,76 +140,24 @@ protected function _initLogging()
-        if (isset($config->log->filename)) {
-            $logFilename = $config->log->filename;
-        } else {
-            $logFilename = 'opus.log';
-            if (! array_key_exists('SERVER_PROTOCOL', $_SERVER)
-                and ! array_key_exists('REQUEST_METHOD', $_SERVER)) {
-                $logFilename = "opus-console.log";
-            }
-        }
-
-        $logfilePath = $config->workspacePath . '/log/' . $logFilename;
-
-        $logfile = @fopen($logfilePath, 'a', false);
-
-        if ($logfile === false) {
-            $path = dirname($logfilePath);
-
-            if (! is_dir($path)) {
-                throw new Exception('Directory for logging does not exist');
-            } else {
-                throw new Exception('Failed to open logging file:' . $logfilePath);
-            }
-        }
-
-        $GLOBALS['id_string'] = uniqid(); // Write ID string to global variables, so we can identify/match individual runs.
-
-        $format = '%timestamp% %priorityName% (%priority%, ID '.$GLOBALS['id_string'].'): %message%' . PHP_EOL;
-        $formatter = new \Zend_Log_Formatter_Simple($format);
-
-        $writer = new \Zend_Log_Writer_Stream($logfile);
-        $writer->setFormatter($formatter);
-
-        $logger = new \Zend_Log($writer);
-        $logLevelName = 'INFO';
-        $logLevelNotConfigured = false;
-
-        if (isset($config->log->level)) {
-            $logLevelName = strtoupper($config->log->level);
-        } else {
-            $logLevelNotConfigured = true;
-        }
-
-        $zendLogRefl = new \ReflectionClass('Zend_Log');
-
-        $invalidLogLevel = false;
-
-        $logLevel = $zendLogRefl->getConstant($logLevelName);
-
-        if (empty($logLevel)) {
-            $logLevel = Zend_Log::INFO;
-            $invalidLogLevel = true;
-        }
-
-        // filter log output
-        $priorityFilter = new \Zend_Log_Filter_Priority($logLevel);
-        \Zend_Registry::set('LOG_LEVEL', $logLevel);
-        $logger->addFilter($priorityFilter);
-
-        if ($logLevelNotConfigured) {
-            $logger->warn('Log level not configured, using default \'' . $logLevelName . '\'.');
-        }
-
-        if ($invalidLogLevel) {
-            $logger->err('Invalid log level \'' . $logLevelName .
-                    '\' configured.');
-        }
-
-        \Zend_Registry::set('Zend_Log', $logger);
-
-        $logger->debug('Logging initialized');
+         if (isset($config->log->filename)) {
+             $logFilename = $config->log->filename;
+         } else {
+             $logFilename = 'opus.log';
+             if (! array_key_exists('SERVER_PROTOCOL', $_SERVER)
+                 and ! array_key_exists('REQUEST_METHOD', $_SERVER)) {
+                 $logFilename = "opus-console.log";
+             }
+         }
+
+        $logService = LogService::getInstance();
+        //        $logger = $logService->setDefaultLog($logFilename);
+
+        $logService->openLog('Zend_Config', $logFilename);
```

Comment:
I think this should be more like
$logger = $logService->createLog('default', $logFilename);
With the log filename optional, but necessary here, because we want to (maybe) change it from the default "opus.log".

The next line with setLog() is hard to understand and should not be necessary. It behaves like getLog while the name is setLog?

Reply:
Maybe a better name would be addLog. Hmm, not sure yet. I guess it depends, maybe we will have both.